ckanext/switzerland/helpers/format_utils.py

```diff
@@ -95,10 +87,43 @@ def prepare_formats_for_index(resources, format_mapping):
 
 
 # all formats that need to be mapped have to be entered in the mapping.yaml
-def _map_to_valid_format(resource_format, format_mapping):
-    resource_format_lower = resource_format.lower()
-    for key, values in format_mapping.iteritems():
-        if resource_format_lower in values:
-            return key
-    else:
-        return None
+def _map_to_valid_format(format):
+    """check whether the format if in the mapping:
+    either as a key or as a value or if it can be derived
+    after cleaning the input format string"""
+    if format in format_mapping.keys():
+        return format
+
+    mappable_formats = format_mapping.values()
+    if format in mappable_formats:
+        return list(mappable_formats).index(format)
+
+    format = _get_cleaned_format(format)
+    if format in mappable_formats:
+        return list(mappable_formats).index(format)
```

Comment:
I don't quite get how this works. `format_mapping.values` should produce a list of lists, right? Something like this:
```
dict_values([['api', 'rest', 'rest api'], ['csv', 'text (.csv)', 'comma ...', 'text (.csv), comma separated'], ...])
```
So if `format` is, say, `api`, it shouldn't be found in `mappable_formats`.

Also, I think `list(mappable_formats).index(format)` should return the index of `format` in the list, as an integer, not a string with the mapped format in it.

I can see the tests are passing so I'm really just confused about what I'm missing here. :-/

Reply:
Thanks for catching this. It was actually a mistake and I don't really see how the tests could pass.